python/tank/authentication/session_cache.py

```diff
@@ -223,29 +224,37 @@ def _try_load_global_authentication_file(file_path):
     return content
 
 
-def _insert_or_update_user(users_file, login, session_token):
+def _insert_or_update_user(users_file, login, session_token, cookies):
     """
     Finds or updates an entry in the users file with the given login and
     session token.
 
     :param users_file: Users dictionary to update.
     :param login: Login of the user to update.
     :param session_token: Session token of the user to update.
+    :param cookies: Session cookies of the user to update.
 
     :returns: True is the users dictionary has been updated, False otherwise.
     """
     # Go through all users
     for user in users_file[_USERS]:
         # If we've matched what we are looking for.
         if _is_same_user(user, login):
+            result = False
             # Update and return True only if something changed.
             if user[_SESSION_TOKEN] != session_token:
                 user[_SESSION_TOKEN] = session_token
-                return True
-            else:
-                return False
+                result = True
+            if user[_COOKIES] != cookies:
+                user[_COOKIES] = cookies
+                result = True
+            return result
     # This is a new user, add it to the list.
-    users_file[_USERS].append({_LOGIN: login, _SESSION_TOKEN: session_token})
+    users_file[_USERS].append({
+        _LOGIN: login,
+        _SESSION_TOKEN: session_token,
+        _COOKIES: cookies
```

Comment:
First, the name... These are actual Web Cookies... Nothing specific to SSO or SAML. Or at least, those cookies that are closely tied to the authentication mechanism are not known on the client side. I don't mind using something else, any suggestions ?  (using SSO_COOKIES seems too specific)